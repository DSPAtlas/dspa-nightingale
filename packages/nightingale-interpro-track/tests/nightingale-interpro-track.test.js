import "../dist/index";
import * as iproData from "./mockData/interpro-IPR016039.json";
import * as contributors from "./mockData/interpro-contributors.json";
import * as residues from "./mockData/interpro-residues.json";

let rendered;

describe("nightingale-interpro-track tests", () => {
  beforeEach(async () => {
    document.documentElement.innerHTML = `<nightingale-interpro-track length="450" height="20"></nightingale-interpro-track>`;
    rendered = document.querySelector("nightingale-interpro-track");
    await new Promise((resolve) => requestAnimationFrame(() => resolve()));
    await rendered.updateComplete;
    rendered.data = iproData;
    contributors[0].residues = residues;
    rendered.contributors = contributors;
  });

  afterEach(() => {
    document.querySelector("nightingale-interpro-track").remove();
  });

  test("it should show nightingale-interpro-track and the current number of residue features coverage mask rects and no highlight", () => {
    expect(document.querySelector("nightingale-interpro-track")).toBeDefined();
    expect(
      document.querySelectorAll("path.feature.rectangle.residue").length
    ).toBe(16);
    expect(document.querySelectorAll("rect").length).toBe(13);
    expect(document.querySelector("g.highlighted>rect")).toBeFalsy();
  });

  test("it should zoom in with correct number of features and location groups and no highlight", (done) => {
    rendered.setAttribute("display-start", "260");
    rendered.setAttribute("display-end", "264");
    window.requestAnimationFrame(() => {
      expect(
        document.querySelectorAll("path.rectangle.child-fragment.feature")
          .length
      ).toBe(6);
      expect(document.querySelectorAll("g.location-group").length).toBe(2);
      expect(document.querySelector("g.highlighted>rect")).toBeFalsy();
      done();
    });
  });

  // eslint-disable-next-line jest/no-disabled-tests
  test.skip("it should display the highlight", (done) => {
    rendered.setAttribute("highlight", "10:30");
    window.requestAnimationFrame(() => {
      expect(document.querySelector("g.highlighted>rect")).toBeTruthy();
      done();
    });
  });
});
