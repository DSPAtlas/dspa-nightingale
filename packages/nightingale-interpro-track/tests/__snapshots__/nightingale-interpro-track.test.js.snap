// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nightingale-interpro-track tests it should display the sequence correctly after highlight 1`] = `
<nightingale-interpro-track
  height="20"
  highlight="10:30"
  length="450"
  style="width: 100%; display: inline-block; line-height: 0;"
>
  <!---->
  <svg
    class="container"
    height="26"
    width="100"
  >
    <g
      class="sequence-features"
    >
      <g
        class="feature-group"
        id="g_IPR016039"
        mask="url(#mask-IPR016039)"
      >
        <g
          class="location-group"
          style="cursor: pointer;"
        >
          <line
            class="cover"
            stroke="#84862d"
            visibility="visible"
            x1="10"
            x2="10"
            y1="12"
            y2="12"
          />
          <path
            class="rectangle feature"
            d="M0,0L0,0L0,16L0,16Z"
            fill="#84862d"
            opacity="1"
            style="stroke: #84862d; pointer-events: auto;"
            transform="translate(10,4)"
          />
          <text
            class="feature-label"
            dominant-baseline="middle"
            fill="black"
            font-size="14"
            style="cursor: inherit; pointer-events: none;"
            text-anchor="middle"
            x="10"
            y="12"
          />
        </g>
        <g
          class="location-group"
          style="cursor: pointer;"
        >
          <line
            class="cover"
            stroke="#84862d"
            visibility="visible"
            x1="10"
            x2="10"
            y1="12"
            y2="12"
          />
          <path
            class="rectangle feature"
            d="M0,0L0,0L0,16L0,16Z"
            fill="#84862d"
            opacity="1"
            style="stroke: #84862d; pointer-events: auto;"
            transform="translate(10,4)"
          />
          <text
            class="feature-label"
            dominant-baseline="middle"
            fill="black"
            font-size="14"
            style="cursor: inherit; pointer-events: none;"
            text-anchor="middle"
            x="10"
            y="12"
          />
        </g>
      </g>
    </g>
    <g
      class="children-features"
    >
      <g
        class="children-group"
        visibility="hidden"
      >
        <g
          class="child-group"
        >
          <g
            class="child-location-group clg-0"
          >
            <line
              class="cover"
              stroke="#bbd279"
              visibility="hidden"
              x1="10"
              x2="10"
              y1="12"
              y2="12"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
          </g>
          <g
            class="child-location-group clg-1"
          >
            <line
              class="cover"
              stroke="#bbd279"
              visibility="hidden"
              x1="10"
              x2="10"
              y1="12"
              y2="12"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
          </g>
          <g
            class="residues-group"
          >
            <g
              class="residues-locations"
              visibility="hidden"
            >
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
            </g>
            <g
              class="residues-locations"
              visibility="hidden"
            >
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
            </g>
          </g>
        </g>
        <g
          class="child-group"
        >
          <g
            class="child-location-group clg-0"
          >
            <line
              class="cover"
              stroke="#87c59c"
              visibility="hidden"
              x1="10"
              x2="10"
              y1="12"
              y2="12"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
          </g>
          <g
            class="child-location-group clg-1"
          >
            <line
              class="cover"
              stroke="#87c59c"
              visibility="hidden"
              x1="10"
              x2="10"
              y1="12"
              y2="12"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
          </g>
        </g>
      </g>
    </g>
    <defs>
      <mask
        id="mask-IPR016039"
      >
        <rect
          fill="white"
          height="2"
          opacity="0.23076923076923075"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="1"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="1"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="1"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.23076923076923075"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="1"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.23076923076923075"
          width="0"
          x="10"
          y="0"
        />
      </mask>
    </defs>
    <g
      class="highlighted"
    >
      <rect
        fill="#FFEB3B66"
        height="26"
        style="pointer-events: none;"
        width="0"
        x="10"
      />
    </g>
    <g
      class="margin"
    >
      <rect
        class="margin-left"
        fill="#FFFFFFDD"
        height="26"
        style="pointer-events: none;"
        width="10"
        x="0"
      />
      <rect
        class="margin-right"
        fill="#FFFFFFDD"
        height="26"
        style="pointer-events: none;"
        width="10"
        x="90"
      />
      <rect
        class="margin-top"
        fill="#FFFFFFDD"
        height="2"
        style="pointer-events: none;"
        width="100"
        x="0"
        y="0"
      />
      <rect
        class="margin-bottom"
        fill="#FFFFFFDD"
        height="2"
        style="pointer-events: none;"
        width="100"
        x="0"
        y="24"
      />
    </g>
  </svg>
</nightingale-interpro-track>
`;

exports[`nightingale-interpro-track tests it should display the track correctly 1`] = `
<nightingale-interpro-track
  height="20"
  length="450"
  style="width: 100%; display: inline-block; line-height: 0;"
>
  <!---->
  <svg
    class="container"
    height="26"
    width="100"
  >
    <g
      class="sequence-features"
    >
      <g
        class="feature-group"
        id="g_IPR016039"
      >
        <g
          class="location-group"
          style="cursor: pointer;"
        >
          <line
            class="cover"
          />
          <path
            class="rectangle feature"
          />
          <text
            class="feature-label"
            dominant-baseline="middle"
            style="cursor: inherit; pointer-events: none;"
            text-anchor="middle"
          />
        </g>
        <g
          class="location-group"
          style="cursor: pointer;"
        >
          <line
            class="cover"
          />
          <path
            class="rectangle feature"
          />
          <text
            class="feature-label"
            dominant-baseline="middle"
            style="cursor: inherit; pointer-events: none;"
            text-anchor="middle"
          />
        </g>
      </g>
    </g>
    <g
      class="children-features"
    >
      <g
        class="children-group"
      >
        <g
          class="child-group"
        >
          <g
            class="child-location-group clg-0"
          >
            <line
              class="cover"
            />
            <path
              class="rectangle child-fragment feature"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
            />
          </g>
          <g
            class="child-location-group clg-1"
          >
            <line
              class="cover"
            />
            <path
              class="rectangle child-fragment feature"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
            />
          </g>
          <g
            class="residues-group"
          >
            <g
              class="residues-locations"
            >
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
            </g>
            <g
              class="residues-locations"
            >
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                transform="translate(10,25)"
              />
            </g>
          </g>
        </g>
        <g
          class="child-group"
        >
          <g
            class="child-location-group clg-0"
          >
            <line
              class="cover"
            />
            <path
              class="rectangle child-fragment feature"
            />
            <path
              class="rectangle child-fragment feature"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
            />
          </g>
          <g
            class="child-location-group clg-1"
          >
            <line
              class="cover"
            />
            <path
              class="rectangle child-fragment feature"
            />
            <path
              class="rectangle child-fragment feature"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
            />
          </g>
        </g>
      </g>
    </g>
    <defs>
      <mask
        id="mask-IPR016039"
      >
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
        <rect
          height="2"
          y="0"
        />
      </mask>
    </defs>
    <g
      class="highlighted"
    />
    <g
      class="margin"
    />
  </svg>
</nightingale-interpro-track>
`;

exports[`nightingale-interpro-track tests it should zoom in 1`] = `
<nightingale-interpro-track
  display-end="264"
  display-start="260"
  height="20"
  length="450"
  style="width: 100%; display: inline-block; line-height: 0;"
>
  <!---->
  <svg
    class="container"
    height="26"
    width="100"
  >
    <g
      class="sequence-features"
    >
      <g
        class="feature-group"
        id="g_IPR016039"
        mask="url(#mask-IPR016039)"
      >
        <g
          class="location-group"
          style="cursor: pointer;"
        >
          <line
            class="cover"
            stroke="#84862d"
            visibility="visible"
            x1="10"
            x2="10"
            y1="12"
            y2="12"
          />
          <path
            class="rectangle feature"
            d="M0,0L0,0L0,16L0,16Z"
            fill="#84862d"
            opacity="1"
            style="stroke: #84862d; pointer-events: auto;"
            transform="translate(10,4)"
          />
          <text
            class="feature-label"
            dominant-baseline="middle"
            fill="black"
            font-size="14"
            style="cursor: inherit; pointer-events: none;"
            text-anchor="middle"
            x="10"
            y="12"
          />
        </g>
        <g
          class="location-group"
          style="cursor: pointer;"
        >
          <line
            class="cover"
            stroke="#84862d"
            visibility="visible"
            x1="10"
            x2="10"
            y1="12"
            y2="12"
          />
          <path
            class="rectangle feature"
            d="M0,0L0,0L0,16L0,16Z"
            fill="#84862d"
            opacity="1"
            style="stroke: #84862d; pointer-events: auto;"
            transform="translate(10,4)"
          />
          <text
            class="feature-label"
            dominant-baseline="middle"
            fill="black"
            font-size="14"
            style="cursor: inherit; pointer-events: none;"
            text-anchor="middle"
            x="10"
            y="12"
          />
        </g>
      </g>
    </g>
    <g
      class="children-features"
    >
      <g
        class="children-group"
        visibility="hidden"
      >
        <g
          class="child-group"
        >
          <g
            class="child-location-group clg-0"
          >
            <line
              class="cover"
              stroke="#bbd279"
              visibility="hidden"
              x1="10"
              x2="10"
              y1="12"
              y2="12"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
          </g>
          <g
            class="child-location-group clg-1"
          >
            <line
              class="cover"
              stroke="#bbd279"
              visibility="hidden"
              x1="10"
              x2="10"
              y1="12"
              y2="12"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
          </g>
          <g
            class="residues-group"
          >
            <g
              class="residues-locations"
              visibility="hidden"
            >
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
            </g>
            <g
              class="residues-locations"
              visibility="hidden"
            >
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
              <path
                class="feature rectangle residue"
                d="M0,0L0,0L0,4L0,4Z"
                fill="white"
                stroke="transparent"
                style="pointer-events: none; stroke: none; opacity: 0.4;"
                transform="translate(10,10)"
              />
            </g>
          </g>
        </g>
        <g
          class="child-group"
        >
          <g
            class="child-location-group clg-0"
          >
            <line
              class="cover"
              stroke="#87c59c"
              visibility="hidden"
              x1="10"
              x2="10"
              y1="12"
              y2="12"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
          </g>
          <g
            class="child-location-group clg-1"
          >
            <line
              class="cover"
              stroke="#87c59c"
              visibility="hidden"
              x1="10"
              x2="10"
              y1="12"
              y2="12"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <path
              class="rectangle child-fragment feature"
              d="M0,0L0,0L0,10L0,10Z"
              fill="white"
              opacity="0.4"
              style="stroke: none; pointer-events: none;"
              transform="translate(10,7)"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
            <text
              class="child-label"
              dominant-baseline="middle"
              fill="black"
              font-size="10"
              style="cursor: default; pointer-events: none;"
              text-anchor="middle"
              x="10"
              y="12"
            />
          </g>
        </g>
      </g>
    </g>
    <defs>
      <mask
        id="mask-IPR016039"
      >
        <rect
          fill="white"
          height="2"
          opacity="0.23076923076923075"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="1"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="1"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="1"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.23076923076923075"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="1"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.6153846153846154"
          width="0"
          x="10"
          y="0"
        />
        <rect
          fill="white"
          height="2"
          opacity="0.23076923076923075"
          width="0"
          x="10"
          y="0"
        />
      </mask>
    </defs>
    <g
      class="highlighted"
    />
    <g
      class="margin"
    >
      <rect
        class="margin-left"
        fill="#FFFFFFDD"
        height="26"
        style="pointer-events: none;"
        width="10"
        x="0"
      />
      <rect
        class="margin-right"
        fill="#FFFFFFDD"
        height="26"
        style="pointer-events: none;"
        width="10"
        x="90"
      />
      <rect
        class="margin-top"
        fill="#FFFFFFDD"
        height="2"
        style="pointer-events: none;"
        width="100"
        x="0"
        y="0"
      />
      <rect
        class="margin-bottom"
        fill="#FFFFFFDD"
        height="2"
        style="pointer-events: none;"
        width="100"
        x="0"
        y="24"
      />
    </g>
  </svg>
</nightingale-interpro-track>
`;
